### PR TITLE
CH embeds platform details to enable/disable some tests

### LINKED ISSUE #1397: [Slow] [Linux Branch] {x64,x86} x {debug,test} Test failure: TypedArrayPrototype
https://github.com/Microsoft/ChakraCore/wiki/Build-Status-%28linux%29#builds-with-slow-tests

```
16:19:06 1>ERROR: Test failed to run correctly: pass not found in output file ():
16:19:06 1>    ch.exe    -DumpOnCrash -WERExceptionSupport -ExtendedErrorStackForTestHost -BaselineMode -maxInterpretCount:1 -maxSimpleJitRunCount:1 -bgjit- -dynamicprofilecache:profile.dpl.UnnamedTest1675   -es6tostringtag -ES6ObjectLiterals -ES6Species -args summary -endargs D:\j\workspace\daily_slow_x8---e1770805\test\es6\ES6TypedArrayExtensions.js >testout1615 2>&1
16:19:06 1>ERROR: name of output file: D:\j\workspace\daily_slow_x8---e1770805\test\es6\testout1615; size: 350; creation: Fri Aug  5 16:14:58 2016, last access: Fri Aug  5 16:14:58 2016, now: Fri Aug  5 16:14:59 2016
16:19:06 1>ERROR: bad output file follows ============
16:19:06 1>*** Running test #12 (11): %TypedArrayPrototype%[@@toStringTag] behavior
16:19:06 1>Test threw exception: assert.throws failed: expected: TypeError 'this' is not a typed array object, actual: <no exception>: %TypedArrayPrototype%[@@toStringTag] throws TypeError if called with no parameter
16:19:06 1>FAILED
16:19:06 1>Summary of tests: total executed: 67; passed: 66; failed: 1
16:19:06 1>ERROR: end of bad output file  ============
```

How to repro locally: https://github.com/Microsoft/ChakraCore/wiki/Jenkins-Repro-Steps#windows-daily

### PR DESCRIPTION
**Pull request by obastemur**
See comments on disabled test case

Attempt to fix #1397

### CODE REVIEW

File: test/fieldopts/objtypespec-add.js [hunk 1]
```diff
@@ -180,7 +180,9 @@ test5();
 test5();
 test5();
 function printAll(n, v) {
-  for (var c in v)
-    printAll(n, v[c]);
+  for (var c in v) {
+    if (typeof v != "string")
+      printAll(c, v[c]);
+  }
```
Comment (ianwjhalliday) [inline test/fieldopts/objtypespec-add.js]: Is this related?

**Comment by ianwjhalliday** [comment]
lgtm

File: test/fieldopts/objtypespec-add.js (hunk 1 above)
Comment (obastemur) [inline test/fieldopts/objtypespec-add.js]: Yep. Platform.OS and ARCH are being String, that loop becomes infinite

**Comment by obastemur** [comment]
@ianwjhalliday Thanks for the review

**Pull request merged**